content/shmem_team_get_config.tex

```diff
@@ -5,18 +5,21 @@
 \begin{apidefinition}
 
 \begin{Csynopsis}
-int @\FuncDecl{shmem\_team\_get\_config}@(shmem_team_t team, shmem_team_config_t *config);
+int @\FuncDecl{shmem\_team\_get\_config}@(shmem_team_t team, long config_mask, shmem_team_config_t *config);
 \end{Csynopsis}
 
 \begin{apiarguments}
   \apiargument{IN}{team}{An \openshmem team handle.}
+  \apiargument{IN}{config\_mask}{
+    The bitwise mask representing the set of configuration parameters to fetch from the given team.
+    }
   \apiargument{OUT}{config}{
     A pointer to the configuration parameters for the given team.}
 \end{apiarguments}
 
 \apidescription{
 \FUNC{shmem\_team\_get\_config} returns through the \VAR{config} argument
-the configuration parameters of the given team, which were assigned according
+the configuration parameters as described by the mask, which were assigned according
 to input configuration parameters when the team was created.
```

Comment:
According to this, the implementation will always need to return the exact values provided when the team was created. It is not able to return the values actually used by the implementation. Are we /really/ sure this is what we want? There is no opportunity for the implementation to provide feedback in this case.

Reply:
I don't think the new description changes previous meaning, at least as I read it.  So, I don't think there is semantics change.

Reply:
Agree you are not changing semantics, but we also did not spend any time discussing this specific semantic in the teams proposal. Is what we have now the right thing?

Reply:
I think it is useful to return the values actually used by implementation. As I read it, I don't think the above prevents implementation from returning actually used values. Do you suggest to be more explicit about this ?

Reply:
Given the pain we just went through in the MPI Forum to clarify this exact semantic -- yes, I strongly encourage not leaving this open to interpretation. You could look at the text I wrote for MPI info in the [2019 draft](https://www.mpi-forum.org/) on page 271 under `MPI_Comm_get_info`. The text took several revision cycles before we got it right.